tests/test_snowflake_common.py

```diff
@@ -243,7 +247,7 @@ def test_get_slice_with_limit_with_offset_not_enough_data(
     ds: DataSlice = SnowflakeCommon().get_slice(connect, snowflake_datasource, offset=5, limit=5)
     assert result.call_count == 3
     assert len(ds.df) == 1
-    assert ds.stats.total_returned_rows == 1
+    assert ds.pagination_info.pagination_info.total_rows == 6
```

Comment:
mhhh, please can you add comments explaining why we're having `6` here ?

Reply:
253521b777b710b4a17adfe1dc517e2b61ed2040